app/models/manageiq/providers/openstack/inventory/parser/infra_manager.rb

```diff
@@ -0,0 +1,366 @@
+      # https://github.com/fog/fog-openstack/issues/197
+      :provision_state      => host.provision_state.nil? ? "available" : host.provision_state,
+    }
+  end
+
+  def parse_host_disks(extra_attributes)
+    return [] if extra_attributes.nil? || (disks = extra_attributes.fetch_path('disk')).blank?
+
+    disks.keys.delete_if { |x| x.include?('{') || x == 'logical' }.map do |disk|
+      # Logical index contains number of logical disks
+      # TODO(lsmola) For now ignoring smart data, that are in format e.g. sda{cciss,1}, we need to design
+      # how to represent RAID
+      # Convert the disk size from GB to B
+      disk_size = disks.fetch_path(disk, 'size').to_i * 1_024**3
+      {
+        :device_name     => disk,
+        :device_type     => 'disk',
+        :controller_type => 'scsi',
+        :present         => true,
+        :filename        => disks.fetch_path(disk, 'id') || disks.fetch_path(disk, 'scsi-id'),
+        :location        => disk,
+        :size            => disk_size,
+        :disk_type       => nil,
+        :mode            => 'persistent'
+      }
+    end
+  end
+
+  def server_address(server, key)
+    # TODO(lsmola) Nova is missing information which address is primary now,
+    # so just taking first. We need to figure out how to identify it if
+    # there are multiple.
+    server&.addresses&.fetch_path('ctlplane', 0, key)
+  end
+
+  def get_purpose(instance_uuid)
+    collector.stack_resources_by_id[instance_uuid]&.dig('resource_name')
+  end
+
+  def identify_product(instance_uuid)
+    purpose = get_purpose(instance_uuid)
+    return nil unless purpose
+
+    if purpose == 'NovaCompute'
+      'rhel (Nova Compute hypervisor)'
+    else
+      "rhel (No hypervisor, Host Type is #{purpose})"
+    end
+  end
+
+  def identify_host_name(instance_uuid, uid)
+    purpose = get_purpose(instance_uuid)
+    return uid unless purpose
+
+    "#{uid} (#{purpose})"
+  end
+
+  def identify_primary_mac_address(host)
+    server = collector.servers_by_id[host.instance_uuid]
+    server_address(server, 'OS-EXT-IPS-MAC:mac_addr')
+  end
+
+  def identify_primary_ip_address(host)
+    server = collector.servers_by_id[host.instance_uuid]
+    server_address(server, 'addr')
+  end
+
+  def identify_ipmi_address(host)
+    host.driver_info["ipmi_address"]
+  end
+
+  def identify_hypervisor_hostname(host)
+    collector.servers_by_id[host.instance_uuid]&.name
+  end
+
+  def lookup_power_state(power_state_input)
+    case power_state_input
+    when "power on"               then "on"
+    when "power off", "rebooting" then "off"
+    else                               "unknown"
+    end
+  end
+
+  def lookup_connection_state(power_state_input)
+    case power_state_input
+    when "power on"               then "connected"
+    when "power off", "rebooting" then "disconnected"
+    else                               "disconnected"
+    end
+  end
+
+  def clusters_and_host_mapping
```

Comment:
NOTE: I'd like to move this into the Collector as `collector.clusters` but currently it uses the parsed orchestration stacks (`@data_index[:orchestration-stacks]`).  Once we move this to use the collected stacks instead then we can drop all of the `process_collection`/`@data`/`@data_index` stuff.

That said, this works the same way as it does in classic refresh and I'd like to do this change in a follow-up PR

Reply:
Nevermind I was able to get this working pretty easily